src/main/java/core/basesyntax/BinaryTree.java

```diff
@@ -22,26 +24,51 @@ public void setRoot(Node root) {
     /* Given a binary tree, print its nodes according to the 
       "bottom-up" post order traversal. */
     public List<Node> getNodesPostOrder(Node node) {
-        System.out.println("Please provide your implementation");
-
-        return new ArrayList<>();
+        List<Node> result = new ArrayList<>();
+        if (node.getLeft() != null) {
+            result.addAll(getNodesPostOrder(node.getLeft()));
+        }
+        if (node.getRight() != null) {
+            result.addAll(getNodesPostOrder(node.getRight()));
+        }
+        result.add(node);
+        return result;
     }
 
     /* Given a binary tree, print its nodes in inorder*/
     public List<Node> getNodesInorder(Node node) {
-        System.out.println("Please provide your implementation");
-        return new ArrayList<>();
+        List<Node> result = new ArrayList<>();
+        if (node.getLeft() != null) {
+            result.addAll(getNodesInorder(node.getLeft()));
+        }
+        result.add(node);
+        if (node.getRight() != null) {
+            result.addAll(getNodesInorder(node.getRight()));
+        }
+        return result;
     }
 
     /* Given a binary tree, print its nodes in pre order*/
     public List<Node> getNodesPreOrder(Node node) {
-        System.out.println("Please provide your implementation");
-        return new ArrayList<>();
+        List<Node> result = new ArrayList<>();
+        result.add(node);
+        if (node.getLeft() != null) {
+            result.addAll(getNodesPreOrder(node.getLeft()));
+        }
+        if (node.getRight() != null) {
+            result.addAll(getNodesPreOrder(node.getRight()));
+        }
+        return result;
     }
 
-    /* Given a binary tree, print its nodes in pre order*/
+    /* Given a binary tree, print its nodes in breadth order*/
     public List<Node> getNodesBreadthFirst(Node node) {
-        System.out.println("Please provide your implementation");
-        return new ArrayList<>();
+        return getNodesPreOrder(node).stream()
+                .collect(Collectors.groupingBy(Node::getKey))
+                .entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .map(Map.Entry::getValue)
+                .map(value -> value.get(0))
+                .collect(Collectors.toList());
```

Comment:
unfortunately this doesn't work for different test values
let's think, can we use some kind of queue here?